File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = [0, 0]
    brown = (brown+4)//2
    for i in range(1, brown//2+1):
        j = brown - i
        if (i-2)*(j-2) == yellow:
            answer = [j, i]
            break
    return answer
            

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    
    let brownSum = (brown+4)/2; // 갈색 가로 세로 하나씩 더한 크기
    for (let i=1; i<= Math.floor(brownSum/2); i++) {
        const j = brownSum - i;
        if ((i-2) * (j-2) === yellow){
            return [j, i]
        }
    }

}